File: src/Api/Model/DB/CNAE/CnaeGrupo.cs
using Realms;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CriadorBaseDados.Model.DB.CNAE
{
    /// <summary>
    /// Seção -> Divisão -> Grupo -> Classe -> Subclasse -> Atividade Econômica
    /// </summary>
    public class CnaeGrupo : RealmObject
    {
        [PrimaryKey]
        public int ID { get; set; }
        public string Descricao { get; set; }
        //public IList<string> Observacoes { get;}
        public CnaeDivisao Divisao { get; set; }
        [Backlink(nameof(CnaeClasse.Grupo))]
        public IQueryable<CnaeClasse> Classes { get; }
    }
}

File: src/Mobile/b2b/lib/model/core/cnae/cnae_grupo.dart
import 'cnae_classe.dart';
import 'cnae_divisao.dart';

/// <summary>
/// Seção -> Divisão -> Grupo -> Classe -> Subclasse -> Atividade Econômica
/// </summary>
class CnaeGrupo {
  final int id;
  final String descricao;
  final int codDivisao;
  final List<CnaeClasse> classes;

  CnaeGrupo(this.id, this.descricao, this.codDivisao, this.classes);

  CnaeGrupo.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        descricao = json['descricao'],
        codDivisao = json['codDivisao'],
        classes = json['classes'] ?? List.empty(growable: true);

  Map<String, dynamic> toMap() => <String, dynamic>{
        'id': id,
        'descricao': descricao,
        'codDivisao': codDivisao,
        //'classes': classes,
      };
}
